content/rs/developing/crdbs/developing-streams-crdb.md

```diff
@@ -1,278 +1,265 @@
-| t4   | XREAD COUNT 2 STREAMS x 0 --> [110-0, 120-0]       |                                                    |
-| t5   | --- sync ---                                       | --- sync ---                                       |
-| t6   | XREAD COUNT 2 STREAMS x 120 --> [130-0]            |                                                    |
-| t7   | XREAD STREAMS x 0 --> [110-0, 115-0, 120-0, 130-0] | XREAD STREAMS x 0 --> [110-0, 115-0, 120-0, 130-0] |
+| _t1_   | `XADD x 110 f1 v1`                                   | `XADD x 115 f1 v1`                                   |
+| _t2_   | `XADD x 120 f1 v1`                                   |                                                    |
+| _t3_   | `XADD x 130 f1 v1`                                   |                                                    |
+| _t4_   | `XREAD COUNT 2 STREAMS x 0` <br/>**→ [110-0, 120-0]**       |                                                    |
+| _t5_   | _— Sync —_                                       | _— Sync —_                                       |
+| _t6_   | `XREAD COUNT 2 STREAMS x 120` <br/>**→ [130-0]**            |                                                    |
+| _t7_   | `XREAD STREAMS x 0` <br/>**→[110-0, 115-0, 120-0, 130-0]** | `XREAD STREAMS x 0` <br/>**→[110-0, 115-0, 120-0, 130-0]** |
+
 
-As you can see, XREAD skipped entry 115-0.
+You can use XREAD to reliably consume a stream only if all writes to the stream originate from a single region. Otherwise, you should use XREADGROUP, which always guarantees reliable stream consumption.
 
-## Stream consumer-groups in Active-Active databases
+## Consumer groups in Active-Active databases
 
-You can use consumer groups in Active-Active databases.
+Active-Active databases fully support consumer groups with Redis Streams.
 
-Here is an example of a simple XGROUP case:
+Here is an example of a using XGROUP with an active-active database:
 
-| Time | Location 1                  | Location 2                  |
+| Time | Region 1                  | Region 2                  |
 | ---- | --------------------------- | --------------------------- |
-| t1   | XGROUP CREATE x g1 0        |                             |
-| t2   | --- sync ---                | --- sync ---                |
-| t3   |                             | XGROUP CREATE x g2 0        |
-| t4   | --- sync ---                | --- sync ---                |
-| t5   | XINFO GROUPS x --> [g1, g2] | XINFO GROUPS x --> [g1, g2] |
+| _t1_   | `XGROUP CREATE x g1 0`        |                             |
+| _t2_   | _— Sync —_                | _— Sync —_                |
+| _t3_   |                             | `XGROUP CREATE x g2 0`        |
+| _t4_   | _— Sync —_                | _— Sync —_                |
+| _t5_   | `XINFO GROUPS x` <br/>**→ [g1, g2]** | `XINFO GROUPS x` <br/>**→ [g1, g2]** |
 
-Here is an example of a concurrent XADD case:
+Here is an example of creating two consumer groups concurrently:
```

Comment:
Creating consumer groups concurrently is a nice example, but I think that XADD-ing messages in parallel is much more interesting and useful

Reply:
There are examples of XADDing messages in parallel elsewhere in this doc. This example is in the context of consumer groups specifically.